---
title: "Analysis: THAP1-RNASeq Enrichment"
subtitle: "Reanalysis of Loss-of-function mutations in the dystonia gene THAP1 impair proteasome function by inhibiting PSMB5 expression"
author: "Michael Kaufman - RBI Informatics Fellow"
output: html_document
format:
  html:
    embed-resources: true
    self_contained: true
    code-fold: true
    code-summary: "[code]"
    code-overflow: wrap
page-layout: full
editor: visual
execute:
    freeze: auto
toc	: true
toc_float: true
toc-location: left
---

## Notes

This is a reanalysis of the RNASeq data from the paper "Loss-of-function mutations in the dystonia gene THAP1 impair proteasome function by inhibiting PSMB5 expression". Supplemental Data Table 2. This was pre-processed data from the original paper. Here I want to examine any pathway enrichment in the differentially expressed genes for the THAP1 mutants.

## Environment

```{r setup, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, fig.align='center')
```

```{r}
library(knitr)
library(dplyr)
library(Seurat)
library(patchwork)
library(hdf5r)
library(clustree)

library(tidyr)
library(dplyr)
library(htmltools)
library(DT)

library(AnnotationDbi)
library(org.Hs.eg.db)
library(clusterProfiler)
library(enrichplot)
library(DOSE)
library(pathview)
library(msigdbr)

# make analysis outs dir
dir.create("analysis_outs", showWarnings = FALSE)
```

```{r}
# load xlsx file to data frame ignore first row
library(readxl)
data_og <- read_excel("media-2.xlsx", skip = 1)
head(data_og)
# make the column names more appropriate for R to handle
colnames(data_og) <- make.names(colnames(data_og))
head(data_og)
length(data_og$Gene)
```


```{r}
# filter out non-significant genes by the FDR columns
data <- data_og %>% filter(FDR..sgTHAP1s_vs_sgControls. < 0.05, log2FC.sgTHAP1s_vs_sgControls. > 0.5)
data2 <- data_og %>% filter(FDR..sgTHAP1s_vs_Un. < 0.05, log2FC.sgTHAP1s_vs_Un. > 0.5)
length(data$Gene)
length(data2$Gene)

```
```{r}
# get the gene list for the enrichment analysis
gene_list <- data$Gene
gene_list2 <- data2$Gene

# perform the enrichment analysis
enrich_res <- enrichGO(gene          = gene_list,
                       OrgDb         = org.Hs.eg.db,
                       keyType       = "SYMBOL",
                       ont           = "BP",
                       pAdjustMethod = "BH",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05,
                       readable      = TRUE)

# perform the enrichment analysis
enrich_res2 <- enrichGO(gene          = gene_list2,
                       OrgDb         = org.Hs.eg.db,
                       keyType       = "SYMBOL",
                       ont           = "BP",
                       pAdjustMethod = "BH",
                       pvalueCutoff  = 0.05,
                       qvalueCutoff  = 0.05,
                       readable      = TRUE)
```

```{r}
# plot the results
barplot(enrich_res, showCategory=10) + 
    ggtitle("THAP1 Mutants vs Controls Enrichment Analysis")

```
```{r, fig.height=20}

barplot(enrich_res2, showCategory = 22) + 
    ggtitle("THAP1 Mutants vs Untransfected Enrichment Analysis") + 
    theme(axis.text.y = element_text(size = 20),
          title = element_text(size = 50))

```
```{r}
# show DT datatable of enriched terms
dt_enrich_res2 <- as.data.frame(enrich_res2)
#drop ID column
dt_enrich_res2 <- dt_enrich_res2[, -1]
# move geneID column to be the second column; geneID is the 7th column
dt_enrich_res2 <- dt_enrich_res2[, c(1, 7, 2:6)]

DT::datatable(dt_enrich_res2)
```